test/inject.test.js

```diff
@@ -86,3 +86,32 @@ test('routes correctly the message between two routes', async (t) => {
   t.same(await promise, message)
   ws.terminate()
 })
+
+test('use the upgrade context to upgrade if there is some hook', async (t) => {
+  const fastify = buildFastify(t)
+  const message = 'hi from client'
+
+  let _resolve
+  const promise = new Promise((resolve) => { _resolve = resolve })
+
+  fastify.register(
+    async function (instance) {
+      instance.addHook('preValidation', async (request, reply) => {
+        if (request.headers['api-key'] !== 'some-random-key') {
+          return reply.code(401).send()
+        }
+      })
+
+      instance.get('/', { websocket: true }, function (conn) {
+        conn.once('data', chunk => {
+          _resolve(chunk.toString())
+        })
+      })
+    })
+
+  await fastify.ready()
+  const ws = await fastify.injectWS('/', { headers: { 'api-key': 'some-random-key' } })
+  ws.send(message)
+  t.same(await promise, message)
+  ws.terminate()
+})
```

Comment:
I think this doesn't test the other case where if you pass the wrong API key, you get a 401. Does that work?